### DISCUSSION TITLE
Support gradient for "chapters"

### DISCUSSION THREAD

**Discussion by icpmoles** [sumatrapdfreader/sumatrapdf#4196] [Ideas]
The "GradientColors" function is quite useful already but I think it can be improved if it offered a way to use the PDF bookmarks' as the bounds for the gradient instead of the first/last page of the document.

For example, in a book with multiple chapters you may use the background color to see how much it is left in the chapter instead of the whole book.

Simple graphic to illustrate the present situation
![gradient comparison](https://github.com/sumatrapdfreader/sumatrapdf/assets/8940011/00670e5b-9e31-4b23-ba41-73ab14690f47)

**Comment by GitHubRulesOK** [comment]
whilst good in theory  and possible in a word processor where there are physical divisions per marked section,
in a PDF there is no possible concept without the author used a custom method.

PDF is sub divided as 

possibly PDF within PDF (nested documents)
A PDF document of one or more collections of pages (uncommon as all pages are counted from Zero to N-1 then converted into 1-N)
Each single charted Page positional at any compass point to any other page, theoretically in any order (Each Page is rarely sub divided)

So  there is no division into lines per page nor words nor paragraphs nor chapters nor parts of a single Document or Page
